.github/workflows/reusable_release_automation.yaml

```diff
@@ -0,0 +1,115 @@
+      - name: Check block endpoint
+        run: |
+          http_status=$(curl -s -o /dev/null -w "%{http_code}" "${{ inputs.block_endpoint }}")
+
+          if [ "$http_status" != 200 ]; then
+            echo "Fail checking release block endpoint: ${{ inputs.block_endpoint }}"
+            exit 1
+          fi
+
+  promote_prerelease:
+    needs: check_block_endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Promote previous prerelease to release
+        env:
+          GH_TOKEN: "${{ secrets.bot_token }}"
+        run: |
+          # Get the date of the latest release.
+          LATEST_RELEASE=$(
+            gh release list --json publishedAt,isLatest -R "${{ github.repository }}" --jq '
+              .[] | select(.isLatest) | .publishedAt | fromdateiso8601
+            '
+          )
+
+          # List all releases, filter by prereleases that was made later than the latest release, sort by date,
+          # reverse because the last one is the most recent and only grab the last prerelease
+          NON_PUBLISHED_PRERELEASE=$(
+            gh release list --json name,publishedAt,isPrerelease,isLatest,tagName -R "${{ github.repository }}" --jq '
+              [
+                  .[] | select( .isPrerelease and ((.publishedAt | fromdateiso8601) > '${LATEST_RELEASE}'))
+              ]
+              | sort_by(.publishedAt |= fromdateiso8601) | reverse | first
+            '
+          )
+
+          # Get prerelease name
+          NON_PUBLISHED_PRERELEASE_NAME=$(echo "$NON_PUBLISHED_PRERELEASE" | jq -r '.name')
+
+          # Check specifically for "artifacts-pending" or "pre-release-failure" conditions
+          if [[ "$NON_PUBLISHED_PRERELEASE_NAME" == *"artifacts-pending"* || "$NON_PUBLISHED_PRERELEASE_NAME" == *"pre-release-failure"* ]]; then
+            echo "Prerelease $NON_PUBLISHED_PRERELEASE_NAME is blocked due to 'artifacts-pending' or 'pre-release-failure'."
+            # Setting an environment variable for conditional steps later on
+            echo "SHOULD_NOTIFY=1" >> $GITHUB_ENV
+          else
+            PRERELEASE_TAG=$(echo "$NON_PUBLISHED_PRERELEASE" | jq -r '.tagName')
+
+            # Promote the prerelease to a full release
+            gh release edit "$PRERELEASE_TAG" --repo "${{ github.repository }}" --draft=false --prerelease=false
+
+            echo "SHOULD_NOTIFY=0" >> $GITHUB_ENV
+          fi
+
+      - name: Notify failure via Slack
+        if: env.SHOULD_NOTIFY == '1'
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.slack_token }}
+          slack-channel: ${{ secrets.slack_channel }}
+          slack-text: "❌ `${{ github.event.repository.full_name }}`: [promote prerelease failed](${{ github.server_url }}/${{ github.event.repository.full_name }}/actions/runs/${{ github.run_id }})"
+
+  trigger_prerelease:
+    needs: promote_prerelease
+    uses: ./.github/workflows/reusable_trigger_prerelease.yaml
```

Comment:
We don't rename this to `_trigger_prerelease.yaml` because some integrations may still need to use this directly, right?

Reply:
Yes. As we talked yesterday some integrations might still need to call this instead of the new autotamed prerelease one...